solarforecastarbiter/io/reference_observations/common.py

```diff
@@ -281,12 +282,32 @@ def update_site_observations(api, fetch_func, site, observations,
         to end). If None and no values in the API, use end - 7 day.
     end : pandas.Timestamp or None
         End time to get data for. If None, use now.
+    gaps_only : bool, default False
+        If True, only update periods between start and end where there
+        are data gaps.
     """
     site_observations = [obs for obs in observations if obs.site == site]
     if end is None:
         end = _utcnow()
     if start is None:
         start = get_last_site_timestamp(api, site_observations, end)
+    if gaps_only:
+        for gstart, gend in _find_data_gaps(
+                api, site_observations, start, end):
```

Comment:
would be more readable if you first did `gaps = _find_data...` and then `for gstart, gend in gaps`

Reply:
doing this would also let you put `gaps = [(start, end)]` in the else statement and move the for block below the if/else. then you'd only have one line with `_post_data`. just an idea. not necessarily better.